flow/layers/backdrop_filter_layer.cc

```diff
@@ -9,6 +9,28 @@ namespace flutter {
 BackdropFilterLayer::BackdropFilterLayer(sk_sp<SkImageFilter> filter)
     : filter_(std::move(filter)) {}
 
+void BackdropFilterLayer::Diff(DiffContext* context, const Layer* old_layer) {
+  auto subtree = context->BeginSubtree();
+  auto* prev = reinterpret_cast<const BackdropFilterLayer*>(old_layer);
+  if (!context->IsSubtreeDirty()) {
+    assert(prev);
+    if (filter_ != prev->filter_) {
+      context->MarkSubtreeDirty(old_layer->paint_region());
+    }
+  }
+
+  // Backdrop filter paints everywhere in cull rect
+  auto paint_bounds = filter_->computeFastBounds(context->GetCullRect());
+  if (context->IsSubtreeDirty()) {
+    context->AddPaintRegion(paint_bounds);
```

Comment:
Don't we always `AddPaintRegion` no matter if it's dirty or not?

Reply:
Again, same thing here. If nothing needs to be repainted within paint_bounds, we can leave the backdrop filter alone. No point painting it. However the line right after
```cpp
context->AddReadbackRegion(paint_bounds);
```
means, that if anything (no matter if above or below) within paint_bounds needs to be repainted, the whole backdrop filter will be repainted. As before, it is because we can't partially repaint backdrop filter (because around the edges of clip you don't get same result).